autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnLoadEvaluator.java

```diff
@@ -0,0 +1,180 @@
+                    ClusterUtils.TIME_FORMAT.format((double) remainingTime / TimeUtil.MIN_IN_MS));
+        }
+    }
+
+    protected void pollYarnMetricsAndScaleCluster() {
+        long start = System.currentTimeMillis();
+        try {
+            MDCBuilder.buildMdcContext(cluster);
+
+            StackV4Response stackV4Response = cloudbreakCommunicator.getByCrn(cluster.getStackCrn());
+
+            String hostGroupInstanceType =
+                    stackResponseUtils.getHostGroupInstanceType(stackV4Response, loadAlert.getScalingPolicy().getHostGroup());
+
+            YarnScalingServiceV1Response yarnResponse = yarnMetricsClient
+                    .getYarnMetricsForCluster(cluster, hostGroupInstanceType, stackV4Response.getCloudPlatform());
+
+            Map<String, String> hostFqdnsToInstanceId = stackResponseUtils
+                    .getCloudInstanceIdsForHostGroup(stackV4Response, loadAlert.getScalingPolicy().getHostGroup());
+            yarnResponse.getScaleUpCandidates().ifPresentOrElse(
+                    scaleUpCandidates -> handleScaleUp(hostGroupInstanceType, scaleUpCandidates, hostFqdnsToInstanceId.size()),
+                    () -> {
+                        yarnResponse.getScaleDownCandidates().ifPresent(
+                                scaleDownCandidates -> handleScaleDown(scaleDownCandidates, hostFqdnsToInstanceId));
+                    });
+
+            LOGGER.debug("Finished loadEvaluator for cluster {} in {} ms", cluster.getStackCrn(), System.currentTimeMillis() - start);
+        } catch (Exception ex) {
+            LOGGER.info("Failed to process load alert for Cluster {}, exception {}", cluster.getStackCrn(), ex);
+            eventPublisher.publishEvent(new UpdateFailedEvent(clusterId));
+        } finally {
+            LOGGER.debug("Finished loadEvaluator for cluster {} in {} ms", cluster.getStackCrn(), System.currentTimeMillis() - start);
+        }
+    }
+
+    protected void handleScaleUp(String hostGroupInstanceType, NewNodeManagerCandidates newNMCandidates, Integer existingHostGroupSize) {
+        Integer yarnRecommendedHostGroupCount =
+                newNMCandidates.getCandidates().stream()
+                        .filter(candidate -> candidate.getModelName().equalsIgnoreCase(hostGroupInstanceType))
+                        .findFirst().map(NewNodeManagerCandidates.Candidate::getCount)
+                        .orElseThrow(() ->
+                                new RuntimeException(String.format("Yarn Scaling API Response does not contain recommended host count " +
+                                                "for hostGroupInstanceType %s in Cluster %s, Yarn Response %s",
+                                        hostGroupInstanceType, cluster.getStackCrn(), newNMCandidates)));
+
+        int maxAllowedScaleUp = loadAlert.getLoadAlertConfiguration().getMaxResourceValue() - existingHostGroupSize;
+        maxAllowedScaleUp = maxAllowedScaleUp < 0 ? 0 : maxAllowedScaleUp;
```

Comment:
As a future enhancement, we likely need to introduce a parameter which indicates how many nodes we will add in 1 iteration.
e.g. if YARN says add 200 nodes (which is within max node count) - we should not attempt to add 200 nodes in 1 invocation. Instead add 50. Nexy upscale can add the next 50 and so on.

Reply:
ok.

Reply:
Also noticed that this is already covered somewhere in the Cloudbreak flow. Think the default is 100 at a time.